### PR TITLE
Fix move construction of value from object

### PR DESCRIPTION
**Pull request by sdkrystian**

### CODE REVIEW

File: test/value.cpp [hunk 1]
```diff
@@ -499,6 +499,13 @@ class value_test
                 BOOST_TEST(jv.is_object());
                 BOOST_TEST(*jv.storage() == *sp);
             }
+            {
+                // verify moved from
+                object obj{{"a", 1}, {"a", 1}};
```
Comment (vinniefalco) [inline test/value.cpp]: I would not use duplicate keys

File: test/value.cpp [hunk 2]
```diff
@@ -499,6 +499,13 @@ class value_test
...
+                BOOST_TEST(obj.capacity());
```
Comment (vinniefalco) [inline test/value.cpp]: I would not depend on conversion to bool. In other words, spell this out: `obj.capacity() > 0`

File: test/value.cpp [hunk 3]
```diff
@@ -499,6 +499,13 @@ class value_test
...
+                value jv(std::move(obj), {});
+                BOOST_TEST(!obj.capacity());
```
Comment (vinniefalco) [inline test/value.cpp]: I would spell this out: `obj.capacity() == 0`

Comment (sdkrystian) [reply test/value.cpp]: Copy, will apply.

**Pull request closed without merging**